### DISCUSSION TITLE
Maximum rotations on ferris wheel should be 3

### DISCUSSION THREAD

**Discussion by wolfblake** [OpenRCT2/OpenRCT2#15715] [Feature requests]
The maximum rotations, forwards and backwards, for the ferris wheel is 3, with the default set at 1 rotation.

**Comment by e-foley** [comment]
This is consistent with the original RCT2.  What would you prefer?

**Comment by jensj12** [comment]
An option for even shorter rotations? Even with 1 rotation guests want to get off before the ride is finished. I don't think more than 3 rotations is ever useful. The value you set is the amount of rotations before it loads another seat, so everyone already makes ~16 rotations with rotations set to 1.

**Comment by e-foley** [comment]
In order to do less than one rotation, you'd have to move the wheel by a number of carriages that doesn't share a common factor with 16.  So 1/16, 3/16, 5/16, 7/16, 9/16, 11/16, 13/16, and 15/16 would all be theoretically possible.  Right now, the game does 16*n+1, where n is the number of rotations.

That said, is a change like this really worth the effort?  Maybe it makes sense to review this down the line, but for now I'd rather focus on fixing bugs so that we don't introduce more.

**Comment by vzybilly** [comment]
it might be more that the guests are complaining about it? I'd say that a
minimum of one is correct because that would be when their car makes it
back around but the peeps seem to want off to soon, which I think is more
of the issue. maybe skew the time for that feeling for the Ferris wheel?

**Comment by pizza2004** [comment]
I want to get off Mr. Bones Wild Wheel.

**Comment by e-foley** [comment]
@vzybilly, if you watch the ride, you'll see that "1 rotation" means that the wheel rotates once between loading one pair of passengers and loading the next pair of passengers _in the adjacent carriage_.  In other words, each peep has to stay on the ride for 16 loading cycles before being able to get off the ride.  This process takes a very long time.

**Comment by vzybilly** [comment]
oh, ohh. those poor peeps. I would be sick of it as well after five times
but 16/32/48 is just... murderous. I am for a change to the Ferris wheel.

**Comment by AaronVanGeffen** [comment]
To me, it doesn't make sense to let guests leave a Ferris Wheel between full rotations. Surely they'd have to reach ground level before they can get out? Letting them leave the ride after half a rotation (i.e. when at the top of the ride) seems very unrealistic.

**Comment by vzybilly** [comment]
it does but I feel this isn't how actual Ferris wheel s work in practice. I
think I remember something like load, skip afew, load,... and the rider
only sees 1-3 wheel rotations before it comes around to them getting off to
load the car again... I think the implementation is wrong, in original as
well... though I'm not sure, I don't watch the Ferris wheel work...

**Comment by AaronVanGeffen** [comment]
Not sure exactly how it's implemented at the moment, but I suppose the implementation could be changed to only load a handful guests per rotation, instead of 16. Seems more realistic, too.

**Comment by HaasJona** [comment]
Aaron, I think you are confused. Nobody wants the guests to jump out of the wheel when they are in the air.

The current behaviour is as follows:
- The wheel has 16 cabins with 2 seats each.
- 2 Guests enter the bottom cabin
- The wheel does a full rotation (by default) plus one extra step, so each guest is moved one step further after the full rotation
- Then 2 guests enter the new bottom cabin again and this is repeated indefinitely

That means with the default rotation setting of one, each guest actually gets 17 full rotations. Which is way too much and unrealistic. On many real ferris wheels you only get one rotation. Which could probably be done by allowing a rotation setting of "0" which only moves the wheel one step each boarding cycle.

**Comment by e-foley** [comment]
@AaronVanGeffen, I'm not saying that peeps should exit when they're off the ground.  I'm offering a possible solution where, instead of rotating by 17 seats at a time ("1 revolution" setting), the ride rotates a smaller odd number of seats at a time.  For example, say we do 5 seats at a time.  Then the loading pattern is like this:

1, 6, 11, 16, wraparound to 5, 10, 15, wraparound to 4, 9, 14, wraparound to 3, 8, 13, wraparound to 2, 7, 12, wraparound to 1 again.

Notice that the passenger only makes 5 complete rotations instead of the usual 17.

I'm not saying this is a good solution or that the Ferris Wheel behavior even needs tweaking; I just wanted to clarify my earlier statement.

**Comment by duncanspumpkin** [comment]
Its not going to be changed for a long time so there is no point discussing it in such depth.

**Comment by e-foley** [comment]
@HaasJona, you beat me to it.

Regarding realistic behavior... I can't speak for all parks, but the Ferris wheel at the Santa Cruz Beach Boardwalk works in two phases: In the loading phase, they load two sets of two passengers at a time (4 passengers each) then rotate the wheel just over a full rotation so that they can load the next 4 passengers, rotate again to load the next 4, and ao forth until the ride is full.  Then, for the actual ride phase, they rotate the wheel a bunch of times in a row.  After it's gone around enough, they repeat the loading phase.

This behavior is a little too detailed for an RCT game though, I'd say.

**Comment by pizza2004** [comment]
@duncanspumpkin Eh, I think everyone here understands this is something that will happen in the future, but often it doesn't matter if a discussion is pointless, because we're all human and we like to talk about things when we think about them and care about them.

Honestly, I think the best possible behavior would be something like @e-foley just described, but with the loading cycle sped up somewhat.  And then the rotations controls how many times the ride actually rotates after the loading cycle.

**Comment by IntelOrca** [comment]
Why is always the most trivial things that get the most discussion?

**Comment by pizza2004** [comment]
@IntelOrca I'd guess because it's not overly technical, which means even the people who don't have much idea how to program or what to do about stuff that would currently be considered important could have something useful to say about this.  It could also just be that this is far more interesting than a bug that causes the game not to run or something like that.

**Comment by vzybilly** [comment]
the most trivial can be the most fun. that and we are human, humans do
weird things.

anyway, I could see afew different modes of operation. first, load phase,
move one cabin till full, spin the desired amount of times, repeat. second,
the "skip afew" which is more like what we currently have but with smaller
numbers instead of 16. for this, it will have the rotations turned into a
"loading / unloading skip amount" that goes from one to the17. maybe
limited to numbers that make each cabin used since 16 would cause someone
to get loaded, one full rotation then they get unloaded for someone else to
load into, meaning that only one cabin gets used, though it can still be
quick I think... it would be interesting to play with when the time comes.

**Comment by pizza2004** [comment]
I mean, I guess it wouldn't require us to rebalance the Ferris Wheel as much as just, we could probably add one or two new modes, and leave the original mode alone.

**Comment by Gymnasiast** [comment]
Why are you all discussing when @wolfblake hasn't even clarified what he means? Please let him clarify first.
